packages/frontend/core/src/modules/explorer/views/nodes/tag/index.tsx

```diff
@@ -181,16 +181,27 @@ export const ExplorerTagNode = ({
       dropEffect={handleDropEffectOnTag}
       data-testid={`explorer-tag-${tagId}`}
     >
-      {tagDocIds?.map(docId => (
-        <ExplorerDocNode
-          key={docId}
-          docId={docId}
-          reorderable={false}
-          location={{
-            at: 'explorer:tags:docs',
-          }}
-        />
-      ))}
+      <ExplorerTagNodeDocs tag={tagRecord} />
     </ExplorerTreeNode>
   );
 };
+
+/**
+ * the `tag.pageIds$` has a performance issue,
+ * so we split the tag node children into a separate component,
+ * so it won't be rendered when the tag node is collapsed.
+ */
+export const ExplorerTagNodeDocs = ({ tag }: { tag: Tag }) => {
```

Comment:
do we really need to open tagged docs in sidebar?

Reply:
Of course, its in design